sale_order_type/models/account_move.py

```diff
@@ -48,6 +48,11 @@ def _compute_sale_type_id(self):
                 )
                 if sale_type:
                     record.sale_type_id = sale_type
+            # Add manually field to compute, as using api.depends on its compute function would
+            #  trigger unwanted recomputation of partner_bank_id on vendor bills
+            self.env.add_to_compute(
+                self.env["account.move"]._fields["journal_id"], record
+            )
```

Comment:
Careful, AFAIK, when using `add_to_compute`, the field dependencies aren't recomputed unless you explicitly call `modified`, too.

Reply:
Have you tried to change the depends of `partner_bank_id`field using `commercial_partner_id`instead ?

https://github.com/odoo/odoo/blob/16.0/addons/account/models/account_move.py#L1315